replication-api/src/main/java/com/connexta/replication/queue/Queue.java

```diff
@@ -0,0 +1,107 @@
+ *
+ * <ul>
+ *   <li>a Tactical Ion or DIB site we replicate from/to
+ *   <li>a DDF-based site we are harvesting from or replicating to
+ *   <li>a Regional Ion site to which we are replicating our local intel to
+ *   <li>the local Ion site (typically will be used for delete operations only)
+ */
+public interface Queue {
+  /**
+   * Gets the identifier of the site this queue is for.
+   *
+   * @return the site id this queue is for
+   */
+  public String getSite();
+
+  /**
+   * Gets the broker used to retrieved this queue.
+   *
+   * @return the associated broker
+   */
+  public QueueBroker getBroker();
+
+  /**
+   * Inserts the specified task with the given priority into this queue if it is possible to do so
+   * immediately without violating capacity restrictions.
+   *
+   * @param task the task to be added to the queue
+   * @return <code>true</code> if the task was added to this queue; <code>false</code> otherwise
+   * @throws QueueException if an error occurred while performing the operation
+   */
+  public boolean offer(TaskInfo task) throws QueueException;
+
+  /**
+   * Inserts the specified task with the given priority into this queue, waiting up to the specified
+   * maximum amount of time if necessary for space to become available.
+   *
+   * @param task the task to be added to the queue
+   * @param timeout how long to wait before giving up, in units of <code>unit</code>
+   * @param unit a {@code TimeUnit} determining how to interpret the <code>timeout</code> parameter
+   * @return <code>true</code> if the task was added to this queue; <code>false</code> otherwise
+   * @throws InterruptedException if the current thread was interrupted while waiting to insert the
+   *     task
+   * @throws QueueException if an error occurred while performing the operation
+   */
+  public boolean offer(TaskInfo task, long timeout, TimeUnit unit)
+      throws QueueException, InterruptedException;
+
+  /**
+   * Inserts the specified task with the given priority into this queue, waiting if necessary for
+   * space to become available.
+   *
+   * @param task the task to be added to the queue
+   * @throws InterruptedException if the current thread was interrupted while waiting to insert the
+   *     task
+   * @throws QueueException if an error occurred while performing the operation
+   */
+  public void put(TaskInfo task) throws QueueException, InterruptedException;
+
+  /**
+   * Retrieves the next available highest priority task, waiting if necessary until one becomes
+   * available. The returned task is considered locked by the current thread/worker.
+   *
+   * @return the next available task with the highest priority from this queue
+   * @throws InterruptedException if the current thread was interrupted while waiting for a task to
+   *     be returned
+   * @throws QueueException if an error occurred while performing the operation
+   */
+  public Task take() throws QueueException, InterruptedException;
```

Comment:
Typical queue interfaces makes the distinction because taking means you are willing to wait forever for a task while polling means you are willing to wait for a maximum amount of time before control should be returned to you with no tasks.